lib/tasks/stockit_item_detail_import.rake

```diff
@@ -0,0 +1,120 @@
+PERMITTED_DETAIL_TYPES = %w[computer electrical computer_accessory].freeze
+FIXED_DETAIL_ATTRIBUTES = %w[comp_test_status test_status frequency voltage].freeze
+
+#import data from stockit for item detail and save it in package.
+namespace :stockit do
+  desc 'Load all item detail records from Stockit'
+  task add_stockit_item_details_to_packages: :environment do
+    offset = 0
+    per_page = 1000
+
+    loop do
+      items_json = Stockit::ItemSync.new(nil, offset, per_page).index_with_detail
+      offset += per_page
+      stockit_items = JSON.parse(items_json["items"])
+      bar = RakeProgressbar.new(stockit_items.size)
+      break if stockit_items.blank?
+
+      stockit_items.each do |item|
+        bar.inc
+        package = Package.find_by(stockit_id: item["id"]) if item["id"]
+        TaskMethods.new(item, package).run
+      end
+    end
+    bar.finished
+  end
+end
+
+class TaskMethods
+  attr_accessor :item, :package
+
+  def initialize(item, package)
+    @item = item
+    @package = package
+  end
+
+  def run
+    if package && import_and_save_detail? && update_package_detail?
+      package&.update_columns(
+        detail_id: detail_id,
+        detail_type: detail_type,
+      )
+    end
+  end
+
+  def import_and_save_detail?
+    item["id"] && item["detail_type"] && item["detail_id"]
+  end
+
+  def create_new_record?
+    item["detail_id"].nil? || item["detail_id"].blank?
+  end
+
+  def create_detail_with_attributes
+    case detail_type
+    when "computer"
+      Computer.create(computer_attributes)
+    when "electrical"
+      Electrical.create(electrical_attributes)
+    when "computer_accessory"
+      ComputerAccessory.create(computer_accessory_attributes)
+    end
+  end
+
+  def create_empty_record
+    detail_type.classify&.constantize.create unless detail_type.blank?
```

Comment:
Having `unless detail_type.blank?` feels like overly defensive programming here. In the case where `detail_type` is actually blank then we silently continue and don't create a record even though we assume it was done. Better to remove the `unless ...` block and let the method fail so that we expose any potential issues. "Fail early and fast" is a good principle here.

Reply:
That said, I do appreciate this is a 'single use' task and not going to be needed again, but it would be important to confirm that it works correctly as this is data that is being migrated.